test/integration/grpc_json_transcoder_integration_test.cc

```diff
@@ -273,13 +277,17 @@ TEST_P(GrpcJsonTranscoderIntegrationTest, StreamingPost) {
 }
 
 TEST_P(GrpcJsonTranscoderIntegrationTest, InvalidJson) {
+  // Usually the response would be
+  // "Unexpected token.\n"
+  //    "INVALID_JSON\n"
+  //    "^"
+  // But if there's a short read from upstream the full "INVALID_JSON" string
+  // may not be read.
   testTranscoding<bookstore::CreateShelfRequest, bookstore::Shelf>(
       Http::TestHeaderMapImpl{{":method", "POST"}, {":path", "/shelf"}, {":authority", "host"}},
       R"(INVALID_JSON)", {}, {}, Status(),
       Http::TestHeaderMapImpl{{":status", "400"}, {"content-type", "text/plain"}},
-      "Unexpected token.\n"
-      "INVALID_JSON\n"
-      "^");
+      "Unexpected token.\nI", false);
```

Comment:
What if the partial read is shorter than this? Seems still potentially flakey? Same below?

Reply:
The thing I'm fixing is if upstream->Envoy is a short write.  It has to read enough of "INVALID_JSON" to realize it's not Json, so has to get a least one character, so we're guaranteed upstream "I" is read.

The other fix is that the prior RawConnectionDriver would do one read, so possibly not get the full response from Envoy.  That's fixed by using the shared utility, which now keeps reading until it sees "\r\n\r\n" so we can't end up with a subset of the response like "Unex"

Reply:
If it keeps reading until "\r\n\r\n" do we need this fix here then?

Reply:
Yep.  This change protects against short reads upstream-to-Envoy.   The client reading the full response is orthogonal.

Reply:
Sorry I think I'm being dense. I just want to make sure I understand. Since the client will read until "\r\n\r\n", the change to "Unexpected token.\nI" isn't explicitly needed, right?